includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1445,6 +1449,20 @@ private function is_valid_layout( $tag_spec, $node ) {
 		return true;
 	}
 
+	/**
+	 * Whether the node is inside a mustache template.
+	 *
+	 * @param DOMElement $node The node to examine.
+	 * @return bool Whether the node is inside a valid mustache template.
+	 */
+	private function is_inside_mustache_template( DOMElement $node ) {
+		return (
+			( 'template' === $node->parentNode->nodeName && 'amp-mustache' === $node->parentNode->getAttribute( 'type' ) )
+			||
+			( 'script' === $node->parentNode->nodeName && 'text/plain' === $node->parentNode->getAttribute( 'type' ) && 'amp-mustache' === $node->parentNode->getAttribute( 'template' ) )
```

Comment:
What about if the `$node` is nested inside of another element? See dfe3c1767 for failing test.

Reply:
Something like this is needed in the body of this method:

```php
		if ( ! empty( $this->open_elements['template'] ) ) {
			// @todo Walk up to the first template element and verify it has a amp-mustache template type.
		} elseif ( ! empty( $this->open_elements['script'] ) ) {
			// @todo Walk up to the first script element and verify it text/plain type and template=amp-mustache.
		}
		return false;
```

Reply:
See 2f16a192b79fe8110fd1d7d84728c07a66d785ee